### PR TITLE
parser: add straight_join_syntax support, just treat it as JOIN.

### LINKED ISSUE #4871: straight_join syntax support
```
CREATE TABLE t1 (i INT, j INT);
CREATE TABLE t2 (k INT, j INT);
INSERT INTO t1 VALUES(1, 1);
INSERT INTO t2 VALUES(1, 1);
SELECT * FROM t1 NATURAL JOIN t2;
SELECT * FROM t1 JOIN t2 USING (j);
```

MySQL:

```
mysql> SELECT * FROM t1 STRAIGHT_JOIN t2 WHERE t1.i = t2.k;
+------+------+------+------+
| i    | j    | k    | j    |
+------+------+------+------+
|    1 |    1 |    1 |    1 |
+------+------+------+------+
1 row in set (0.00 sec)
```

TiDB:

```
mysql> SELECT * FROM t1 STRAIGHT_JOIN t2 WHERE t1.i = t2.k;
ERROR 1105 (HY000): line 1 column 33 near " WHERE t1.i = t2.k" (total length 51)
```

refer: https://dev.mysql.com/doc/refman/5.7/en/join.html

>STRAIGHT_JOIN is similar to JOIN, except that the left table is always read before the right table. This can be used for those (few) cases for which the join optimizer processes the tables in a suboptimal order.

What version of TiDB are you using (`tidb-server -V`)?

```
mysql> select tidb_version();
+-----------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: 1.0.0
Git Commit Hash: 3d28900feed11d495a3d62542a56b419565d8c0c
Git Branch: master
UTC Build Time: 2017-10-22 07:50:21 |
+-----------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by winkyao**
Fix #4871 

@coocood @shenli @tiancaiamao PTAL

### CODE REVIEW

**Comment by winkyao** [comment]
>STRAIGHT_JOIN is similar to JOIN, except that the left table is always read before the right table. This can be used for those (few) cases for which the join optimizer processes the tables in a suboptimal order.

https://dev.mysql.com/doc/refman/5.7/en/join.html

**Comment by shenli** [comment]
LGTM

File: parser/parser_test.go [hunk 1]
```diff
@@ -362,6 +362,9 @@ func (s *testParserSuite) TestDMLStmt(c *C) {
 		{"select * from t1 natural inner join t2", false},
 		{"select * from t1 natural cross join t2", false},
 
+		// for strait_join
```
Comment (jackysp) [inline parser/parser_test.go]: typo

**Comment by jackysp** [comment]
rest LGTM

**Comment by winkyao** [comment]
@jackysp PTAL

**Pull request merged**